### PR TITLE
Adding support for Timestamps in type inference

### LINKED ISSUE #130: inferSchema option does not infer TimestampType
Version 1.2.0: The inferSchema option does not convert data in the timestamp format YYYY-MM-DD HH:MM:SS to a TimestampType. It keeps it as StringType.

### PR DESCRIPTION
**Pull request by falaki**
Adds support for timestamps and adds some unit tests.

closes #130

### CODE REVIEW

**Comment by falaki** [comment]
@brkyvz would you please take a look at this one too?

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala [hunk 1]
```diff
@@ -419,14 +416,30 @@ class CsvSuite extends FunSuite {
...
+
+  test("Inferring schema") {
+    val results: Array[Row] = new CsvParser()
+      .withDelimiter(',')
+      .withComment('~')
+      .withInferSchema(true)
+      .csvFile(TestSQLContext, commentsFile)
+      .collect()
+
+    val expected =
```
Comment (brkyvz) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: what's the difference between this suite and CsvFastSuite? Not necessary in this PR, but maybe you can define an abstract test suite and have CsvFastSuite and CsvSuite extend it with different configurations?

File: src/test/scala/com/databricks/spark/csv/util/InferSchemaSuite.scala [hunk 2]
```diff
@@ -5,24 +5,33 @@ import org.scalatest.FunSuite
...
     assert(InferSchema.inferField(DoubleType, "test") == StringType)
+    assert(InferSchema.inferField(LongType, "2015-08-20 14:57:00") == TimestampType)
+    assert(InferSchema.inferField(DoubleType, "2015-08-20 15:57:00") == TimestampType)
   }
 
-  test("Type arrays are merged to highest common type"){
+  test("Timestamp field types are inferred correctly from other types") {
+    assert(InferSchema.inferField(IntegerType, "2015-08-20 14") == StringType)
+    assert(InferSchema.inferField(DoubleType, "2015-08-20 14:10") == StringType)
+    assert(InferSchema.inferField(LongType, "2015-08 14:49:00") == StringType)
```
Comment (brkyvz) [inline src/test/scala/com/databricks/spark/csv/util/InferSchemaSuite.scala]: Why are these StringType and not TimestampType?

**Comment by brkyvz** [comment]
LGTM. Just a couple questions

File: src/test/scala/com/databricks/spark/csv/util/InferSchemaSuite.scala (hunk 2 above)
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/util/InferSchemaSuite.scala]: These malformed timestamps (for example they are missing minutes, seconds or dates).

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala (hunk 1 above)
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: Yes, this is a good suggestion. I have been think about doing this and avoid all this code duplication. As you said separate PR.

**Comment by falaki** [comment]
Thanks @brkyvz

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala (hunk 1 above)
Comment (brkyvz) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: In fact allow me! I always wanted to contribute to spark-csv!

**Pull request merged**